jib-core/src/test/java/com/google/cloud/tools/jib/builder/steps/PullBaseImageStepTest.java

```diff
@@ -127,4 +132,49 @@ public void testCall_offlineMode_cached()
 
     Mockito.verify(buildContext, Mockito.never()).newBaseImageRegistryClientFactory();
   }
+
+  @Test
+  public void testObtainPlatformSpecificImageManifest() throws IOException, RegistryException {
+    String manifestListJson =
+        " {\n"
+            + "   \"schemaVersion\": 2,\n"
+            + "   \"mediaType\": \"application/vnd.docker.distribution.manifest.list.v2+json\",\n"
+            + "   \"manifests\": [\n"
+            + "      {\n"
+            + "         \"mediaType\": \"application/vnd.docker.distribution.manifest.v2+json\",\n"
+            + "         \"size\": 424,\n"
+            + "         \"digest\": \"sha256:111111111111111111111111111111111111111111111111111111111111111\",\n"
+            + "         \"platform\": {\n"
+            + "            \"architecture\": \"arm64\",\n"
+            + "            \"os\": \"linux\"\n"
+            + "         }\n"
+            + "      },\n"
+            + "      {\n"
+            + "         \"mediaType\": \"application/vnd.docker.distribution.manifest.v2+json\",\n"
+            + "         \"size\": 425,\n"
+            + "         \"digest\": \"sha256:222222222222222222222222222222222222222222222222222222222222222222\",\n"
+            + "         \"platform\": {\n"
+            + "            \"architecture\": \"targetArchitecture\",\n"
+            + "            \"os\": \"targetOS\"\n"
+            + "         }\n"
+            + "      }\n"
+            + "   ]\n"
+            + "}";
+
+    Mockito.when(buildContext.getContainerConfiguration()).thenReturn(containerConfiguration);
+    Mockito.when(containerConfiguration.getPlatforms())
+        .thenReturn(ImmutableSet.of(new Platform("targetArchitecture", "targetOS")));
+    V22ManifestListTemplate manifestList =
+        JsonTemplateMapper.readJson(manifestListJson, V22ManifestListTemplate.class);
+    ManifestAndDigest<?> manifest = Mockito.mock(ManifestAndDigest.class);
+    Mockito.<ManifestAndDigest<?>>when(
+            registryClient.pullManifest(
+                "sha256:222222222222222222222222222222222222222222222222222222222222222222"))
+        .thenReturn(manifest);
+
+    ManifestAndDigest<?> returnManifest =
+        pullBaseImageStep.obtainPlatformSpecificImageManifest(registryClient, manifestList);
+
+    Assert.assertEquals(manifest, returnManifest);
```

Comment:
nit: `assertEquals` may work, but we can use `assertSame` to check if two object references are same (i.e., if `manifest == returnManifest` evaluates to true, meaning the references are pointing to the same single instance in the memory). OTOH, `assertEquals` may return when two (possible separate) object instances are deemed "equal" (whose definition is given by the class implementation of `.equals()`) even if `manifest != returnManifest`.

Reply:
Good point !! Object comparison requires ` .equals() `